ci/travis-build-gmt.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Build latest GMT master branch
+
+# To return a failure if any commands inside fail
+set -e
+
+export INSTALLDIR="$HOME/gmt-install-dir";
+export COASTLINEDIR="$HOME/gmt-install-dir/coast";
+
+git clone --depth=1 https://github.com/GenericMappingTools/gmt "${HOME}/gmt-master"
+cd "${HOME}/gmt-master"
+bash ci/travis-setup.sh
```

Comment:
@seisman this might be a problem because it uses the Ubuntu packages to compile GMT. So it won't work on Mac. Ideally, we should try to use the conda installed packages but this has causes a lot of trouble for me in the past.

Reply:
@leouieda Currently, the master branch is only built on Linux with Python 3.6 installed. If you really want to test the GMT master branch on Mac, we should use homebrew.

Reply:
OK, do you have any idea how that works? If we can get that setup, I would be willing to ditch the conda-force based jobs in favor of this. It's a lot easier to control and would actually help test GMT itself.

Reply:
I'll try to get this setup in a separate PR

Reply:
@leouieda See this PR https://github.com/GenericMappingTools/gmt/pull/146. 

The only problem is it's too slow, (3 minutes for Linux builds, 13 minutes for Mac builds). 
That's why this PR is not merged into GMT master branch.

If you think it's acceptable, I can add it to gmt-python.